source/tests/system/waveform_helpers.h

```diff
@@ -0,0 +1,109 @@
+#ifndef NIDEVICE_GRPC_TESTS_WAVEFORM_HELPERS
+#define NIDEVICE_GRPC_TESTS_WAVEFORM_HELPERS
+
+#include <fstream>
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include <vector>
+
+namespace ni {
+namespace tests {
+namespace system {
+
+template <typename TComplex>
+struct TestWaveform {
+  double t0;
+  double dt;
+  std::vector<TComplex> data;
+};
+
+template <typename TFloat>
+struct TestIQData {
+  double t0;
+  double dt;
+  std::vector<TFloat> I;
+  std::vector<TFloat> Q;
+};
+
+template <typename TFloat, typename TComplex>
+TComplex complex(TFloat real, TFloat imaginary)
+{
+  auto c = TComplex{};
+  c.set_real(real);
+  c.set_imaginary(imaginary);
+  return c;
+}
+
+template <typename TFloat, typename TComplex>
+std::vector<TComplex> complex_array(
+    std::vector<TFloat> reals,
+    std::vector<TFloat> imaginaries)
+{
+  auto c = std::vector<TComplex>{};
+  c.reserve(reals.size());
+  std::transform(
+      reals.begin(),
+      reals.end(),
+      imaginaries.begin(),
+      std::back_inserter(c),
+      [](TFloat real, TFloat imaginary) { return complex<TFloat, TComplex>(real, imaginary); });
+  return c;
+}
+
+template <typename TFloat>
+TestIQData<TFloat> fetch_iq_data(nlohmann::json json)
```

Comment:
**fetch** is strongly associated with reading data from the driver in rfmx.

Does it work to just call this `load_test_iq_data` and have it be an overload of the file name version? Or `load_xyz_from_json`.?

Reply:
updated this name to `load_iq_data_from_json`